tests/RobotFramework/tests/cumulocity/configuration/configuration_operation.robot

```diff
@@ -0,0 +1,60 @@
+*** Settings ***
+Resource    ../../../resources/common.resource
+Library    Cumulocity
+Library    ThinEdgeIO
+
+Test Tags    theme:c8y    theme:configuration
+Suite Setup    Custom Setup
+Test Teardown    Get Logs
+Test Setup    Test Setup
+
+*** Test Cases ***
+
+Set configuration
+    ${config_url}=    Cumulocity.Create Inventory Binary    config1    CONFIG1    file=${CURDIR}/config1-version2.json
+    ${operation}=    Cumulocity.Set Configuration    CONFIG1    url=${config_url}
+    ${operation}=    Operation Should Be SUCCESSFUL    ${operation}
+    ${target_contents}=    Execute Command    cat /etc/config1.json
+    Should Be Equal    ${target_contents}    {"name":"configuration1","version":2}
+
+Set configuration with broken url
+    ${operation}=    Cumulocity.Set Configuration    CONFIG1    url=invalid://hellö.zip
+    ${operation}=    Operation Should Be FAILED    ${operation}
+    ${target_contents}=    Execute Command    cat /etc/config1.json
+    Should Be Equal    ${target_contents}    {"name":"configuration1"}
+
+Get configuration
+    ${operation}=    Cumulocity.Get Configuration    CONFIG1
+    ${operation}=    Operation Should Be SUCCESSFUL    ${operation}
+    ${events}=    Cumulocity.Device Should Have Event/s    minimum=1    maximum=1    type=CONFIG1
+    ${uploaded_contents}=    Cumulocity.Event Should Have An Attachment    ${events[0]["id"]}
+    ${target_contents}=    Execute Command    cat /etc/config1.json
+    Should Be Equal    ${target_contents}    ${uploaded_contents.decode("utf8")}
+    Should Be Equal    ${target_contents}    {"name":"configuration1"}
+
+Get non existent configuration file
+    Execute Command    rm -f /etc/config1.json
+    File Should Not Exist    /etc/config1.json
+    ${operation}=    Cumulocity.Get Configuration    CONFIG1
+    Operation Should Be FAILED    ${operation}    failure_reason=.*No such file or directory.*
+
+Get non existent configuration type
+    ${operation}=    Cumulocity.Get Configuration    unknown_config
+    Operation Should Be FAILED    ${operation}    failure_reason=.*requested config_type unknown_config is not defined in the plugin configuration file.*
+
+
+*** Keywords ***
+
+Test Setup
+    ThinEdgeIO.Transfer To Device    ${CURDIR}/config1.json         /etc/
+
+Custom Setup
+    ${DEVICE_SN}=    Setup
+    Set Suite Variable    $DEVICE_SN
+    Device Should Exist                      ${DEVICE_SN}
+    ThinEdgeIO.Transfer To Device    ${CURDIR}/c8y-configuration-plugin.toml    /etc/tedge/c8y/
+    ThinEdgeIO.Transfer To Device    ${CURDIR}/config1.json         /etc/
+    ThinEdgeIO.Restart Service    tedge-agent
+    ThinEdgeIO.Disconnect Then Connect Mapper    c8y
```

Comment:
Any specific reason why you're restarting the `tedge-agent` and reconnecting the mapper? Restarting just the plugin would be sufficient for it to pickup the updated config file.

Reply:
Good point. The restarting stuff came out of some other test where the startup sequence was a bit shaky. I will remove those lines, and run the flake finder to check that it works fine without those lines.

Reply:
I removed the unnecessary restarting of agent and mapper, and ran the test 50 times and seems to be stable.

Reply:
> The test looks fine. A few more tests cases that could be added for improved coverage (not necessarily to be added in this PR itself):
> 
> 1. Updates to the `c8y-configuration-plugin.toml` are automatically picked up and the supported config list is updated
> 2. Set non-existent config file with the desired file permissions
> 3. Updating `c8y-configuration-plugin.toml` from the cloud updates the supported config list automatically
> 
> I can think of some more negative test cases like setting broken/invalid `c8y-config-plugin.toml`, setting non-existent config-type etc. But, not sure if they're all relevant here at a system test level.

I just re-read the list of extra test cases and I think it would be definitely worthwhile extending this PR to cover them, thanks for the suggestions.